src/main/java/dev/openfeature/sdk/multiprovider/MultiProvider.java

```diff
@@ -0,0 +1,148 @@
+package dev.openfeature.sdk.multiprovider;
+
+import dev.openfeature.sdk.EvaluationContext;
+import dev.openfeature.sdk.EventProvider;
+import dev.openfeature.sdk.FeatureProvider;
+import dev.openfeature.sdk.Metadata;
+import dev.openfeature.sdk.ProviderEvaluation;
+import dev.openfeature.sdk.Value;
+import dev.openfeature.sdk.exceptions.GeneralError;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.json.JSONObject;
+
+/** <b>Experimental:</b> Provider implementation for Multi-provider. */
+@Slf4j
+public class MultiProvider extends EventProvider {
+
+    @Getter
+    private static final String NAME = "multiprovider";
+
+    public static final int INIT_THREADS_COUNT = 8;
+    private final Map<String, FeatureProvider> providers;
+    private final Strategy strategy;
+    private String metadataName;
```

Comment:
[suggestion] as this variable is more than just a name - it is a json object, we might want to rename it to something more representing like 

```suggestion
    private String metadata;
```